src/user/user.ts

```diff
@@ -124,38 +124,67 @@ export class User implements IUser {
 
   public setWalletInfo(
     id: string,
-    info: string | WalletDescriptor,
-    legacyWallet?: boolean
+    info: string | WalletDescriptor
   ) {
-    let walletInfo = this.getWalletInfo(id, legacyWallet);
+    let walletInfo = this.getWalletInfo(id);
     if (!walletInfo) {
+      const isLegacyWallet = this.isLegacyWalletID(id);
```

Comment:
if we use uid here the value will alway be true, I think

Reply:
Yes, it is okay to true here, we're in inserting mode, we only accept derived HD path
either private key or uid will be failed in this state